bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/handler/RadioThermostatHandler.java

```diff
@@ -0,0 +1,600 @@
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.StateOption;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.radiothermostat.internal.RadioThermostatConfiguration;
+import org.openhab.binding.radiothermostat.internal.RadioThermostatHttpException;
+import org.openhab.binding.radiothermostat.internal.RadioThermostatStateDescriptionProvider;
+import org.openhab.binding.radiothermostat.internal.json.RadioThermostatData;
+import org.openhab.binding.radiothermostat.internal.json.RadioThermostatJsonHumidity;
+import org.openhab.binding.radiothermostat.internal.json.RadioThermostatJsonModel;
+import org.openhab.binding.radiothermostat.internal.json.RadioThermostatJsonName;
+import org.openhab.binding.radiothermostat.internal.json.RadioThermostatJsonResponse;
+import org.openhab.binding.radiothermostat.internal.json.RadioThermostatJsonRuntime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link RadioThermostatHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * Based on the 'airquality' binding by Kuba Wolanin
+ *
+ * @author Michael Lobstein - Initial contribution
+ */
+@NonNullByDefault
+public class RadioThermostatHandler extends BaseThingHandler {
+    private @Nullable final RadioThermostatStateDescriptionProvider stateDescriptionProvider;
+
+    private Logger logger = LoggerFactory.getLogger(RadioThermostatHandler.class);
+
+    private final HttpClient httpClient;
+
+    private static final String URL = "http://%hostName%/%resource%";
+
+    private static final String DEFAULT_RESOURCE = "tstat";
+    private static final String RUNTIME_RESOURCE = "tstat/datalog";
+    private static final String HUMIDITY_RESOURCE = "tstat/humidity";
+    private static final String MODEL_RESOURCE = "tstat/model";
+    private static final String NAME_RESOURCE = "sys/name";
+
+    private static final int DEFAULT_REFRESH_PERIOD = 2;
+    private static final int DEFAULT_LOG_REFRESH_PERIOD = 10;
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private @Nullable ScheduledFuture<?> logRefreshJob;
+
+    private RadioThermostatData rthermData = new RadioThermostatData();
+
+    private Gson gson;
+
+    private int initialized = 0;
+    private int retryCounter = 0;
+
+    public RadioThermostatHandler(Thing thing,
+            @Nullable RadioThermostatStateDescriptionProvider stateDescriptionProvider, HttpClient httpClient) {
+        super(thing);
+        this.stateDescriptionProvider = stateDescriptionProvider;
+        this.httpClient = httpClient;
+        gson = new Gson();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing RadioThermostat handler.");
+        RadioThermostatConfiguration config = getConfigAs(RadioThermostatConfiguration.class);
```

Comment:
calling `getConfigAs` is a costly operation, so you should try to cache the parsed config in a field so that other parts of the code can use it.